p2p/p2p_test.go

```diff
@@ -48,407 +41,337 @@ func startNewService(t *testing.T, cfg *Config, msgSendChan chan []byte, msgRecC
 	return node
 }
 
-func TestBuildOpts(t *testing.T) {
-	tmpPDir, err := ioutil.TempDir(os.TempDir(), "p2p-test")
-	if err != nil {
-		t.Fatal(err)
-	}
-	defer os.RemoveAll(tmpPDir)
-	tmpDir := path.Join(tmpPDir, "data")
-	err = os.Mkdir(tmpDir, os.ModePerm+os.ModeDir)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	testCfgA := &Config{
-		BootstrapNodes: []string{},
-		Port:           7001,
-		DataDir:        tmpDir,
-	}
-
-	_, err = testCfgA.buildOpts()
-	if err != nil {
-		t.Fatalf("TestBuildOpts error: %s", err)
-	}
-
-	if testCfgA.privateKey == nil {
-		t.Error("Private key is nil")
-	}
-
-	testCfgB := &Config{
-		BootstrapNodes: []string{},
-		Port:           7001,
-		DataDir:        tmpDir,
-	}
-
-	_, err = testCfgB.buildOpts()
-	if err != nil {
-		t.Fatalf("TestBuildOpts error: %s", err)
-	}
-
-	if testCfgA.privateKey == testCfgB.privateKey {
-		t.Error("Private key does not match first key generated")
+func TestStartService(t *testing.T) {
+	config := &Config{
+		Port:        7001,
+		RandSeed:    1,
+		NoBootstrap: true, // TODO: fix no bootstrap, this should be required
+		NoMdns:      true, // TODO: investigate failed dials, disable for now
 	}
+	node := startNewService(t, config, nil, nil)
+	node.Stop()
 }
 
-func TestBootstrapConnect(t *testing.T) {
-	bootnodeCfg := &Config{
-		BootstrapNodes: nil,
-		Port:           7000,
-		RandSeed:       1,
-		NoBootstrap:    true,
-		NoMdns:         true,
+func TestBootstrap(t *testing.T) {
+	configA := &Config{
+		Port:        7001,
+		RandSeed:    1,
+		NoBootstrap: true, // TODO: fix no bootstrap, this should be required
+		NoMdns:      true, // TODO: investigate failed dials, disable for now
 	}
 
-	bootnode := startNewService(t, bootnodeCfg, nil, nil)
-	defer bootnode.Stop()
-	bootnodeAddr := bootnode.host.fullAddrs()[0]
+	nodeA := startNewService(t, configA, nil, nil)
+	defer nodeA.Stop()
+
+	addrA := nodeA.host.fullAddrs()[0]
 
-	nodeCfg := &Config{
-		BootstrapNodes: []string{bootnodeAddr.String()},
-		Port:           7001,
+	configB := &Config{
+		BootstrapNodes: []string{addrA.String()},
+		Port:           7002,
 		RandSeed:       2,
-		NoBootstrap:    false,
-		NoMdns:         true,
+		NoMdns:         true, // TODO: investigate failed dials, disable for now
```

Comment:
IMO repeating this everywhere is unnecessary work to remove, opening an issue would be sufficient.

Reply:
Yea, maybe a bit excessive but the intention was to make it clear where to remove when the issue is resolved. Can we keep it for now?

Reply:
Fine to leave as you will surely be the one removing them :)